src/main/java/uk/gov/companieshouse/api/accounts/controller/IntangibleAssetsController.java

```diff
@@ -0,0 +1,138 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.api.accounts.AttributeName;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.links.SmallFullLinkType;
+import uk.gov.companieshouse.api.accounts.model.rest.SmallFull;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.intangible.IntangibleAssets;
+import uk.gov.companieshouse.api.accounts.model.validation.Errors;
+import uk.gov.companieshouse.api.accounts.service.impl.IntangibleAssetsService;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
+import uk.gov.companieshouse.api.accounts.utility.ApiResponseMapper;
+import uk.gov.companieshouse.api.accounts.utility.ErrorMapper;
+import uk.gov.companieshouse.api.accounts.utility.LoggingHelper;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.swing.text.DefaultEditorKit;
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping(value = "/transactions/{transactionId}/company-accounts/{companyAccountId}/small-full/notes/intangible-assets", produces = MediaType.APPLICATION_JSON_VALUE)
+public class IntangibleAssetsController {
+
+    @Autowired
+    private IntangibleAssetsService intangibleAssetsService;
+
+    @Autowired
+    private ApiResponseMapper apiResponseMapper;
+
+    @Autowired
+    private ErrorMapper errorMapper;
+
+    @PostMapping
+    public ResponseEntity create(@RequestBody IntangibleAssets intangibleAssets,
+                                 BindingResult bindingResult,
+                                 @PathVariable("companyAccountId") String companyAccountId,
+                                 HttpServletRequest request) {
+
+        if (bindingResult.hasErrors()) {
+            Errors errors = errorMapper.mapBindingResultErrorsToErrorModel(bindingResult);
+            return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+
+        }
+
+        Transaction transaction = (Transaction) request.getAttribute(AttributeName.TRANSACTION.getValue());
+
+        try {
+            ResponseObject<IntangibleAssets> response = intangibleAssetsService.create(intangibleAssets, transaction,
+                    companyAccountId, request);
+            return apiResponseMapper.map(response.getStatus(), response.getData(), response.getErrors());
+
+        }
+        catch(DataException ex) {
```

Comment:
Nitpicky: Can we have the catch on the same line as the preceding curly brace please?

Reply:
Done